unity/PitayaExample/Assets/Pitaya/PitayaClient.cs

```diff
@@ -19,15 +19,16 @@ public class PitayaClient : IDisposable, IPitayaClient, IPitayaListener
         private bool _disposed;
         private uint _reqUid;
         private Dictionary<uint, Action<string, string>> _requestHandlers;
+        private IPitayaBinding _binding = new PitayaBinding();
         
         public PitayaClient() : this(false) {}
         public PitayaClient(int connectionTimeout) : this(false, null, connectionTimeout: connectionTimeout) {}
         public PitayaClient(string certificateName = null) : this(false, certificateName: certificateName) {}
-        
-        public PitayaClient(bool enableReconnect = false, string certificateName = null, int connectionTimeout = DEFAULT_CONNECTION_TIMEOUT, IPitayaQueueDispatcher queueDispatcher = null)
+        public PitayaClient(bool enableReconnect = false, string certificateName = null, int connectionTimeout = DEFAULT_CONNECTION_TIMEOUT, IPitayaQueueDispatcher queueDispatcher = null, IPitayaBinding binding = null)
         {
+            if (binding != null) _binding = binding;
```

Comment:
Given the assignment in line 22, I guess you will never enter this `if`, right?

Reply:
Kind of, it'll never enter the if clause in production indeed, but this will be used for unit testing, to replace the default implementation by mocks, stubs, etc.

Reply:
Fair enough.